### PR TITLE
Fix language not being recognized correctly

### PR DESCRIPTION
**Pull request by ExplodingDragon**
Support for `en_US` type language formats.

### CODE REVIEW

**Comment by fiftin** [comment]
But your changes break other languages.

File: web/src/plugins/i18.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ Vue.use(VueI18n);
 let locale = localStorage.getItem('lang');
 
 if (!locale) {
-  locale = navigator.language.split('-')[0];
+  locale = navigator.language.replace('-', '_').toLocaleLowerCase();
```
Comment (fiftin) [inline web/src/plugins/i18.js:10]: Looks like "fr", "ru", etc will not work after this change.

Comment (ExplodingDragon) [reply web/src/plugins/i18.js:10]: When an exact match cannot be made, the country code will be removed and the match will continue.

图片

**Comment by ExplodingDragon** [comment]
When an exact match cannot be made, the country code will be removed and the match will continue. The test doesn't seem to be a problem.

ru

图片

fr

图片


zh-cn

图片

File: web/src/App.vue [hunk 2]
```diff
@@ -670,6 +670,11 @@ const LANGUAGES = {
 function getLangInfo(locale) {
   let res = LANGUAGES[locale];
 
+  // failback short i18n
+  if (!res) {
+    res = LANGUAGES[locale.split('_')[0]];
```
Comment (tboerger) [inline web/src/App.vue:675]: This should catch the case mentioned by @fiftin

**Review by tboerger — APPROVED** [review]

**Pull request merged**